### PR TITLE
Adds fixes and tests for Analytics

### PR DESCRIPTION
**Pull request by bmourat**

### CODE REVIEW

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m [hunk 1]
```diff
@@ -227,6 +228,41 @@ - (void)testApplyEnabledStateWorks {
   [[MSAnalytics sharedInstance].sessionTracker stop];
 }
 
+- (void)testApplyEnabledStateWithAutoPageTrackingEnabled {
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: What do you want to test here?

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m [hunk 2]
```diff
@@ -329,6 +365,45 @@ - (void)testTrackEventWithoutProperties {
...
+  OCMStub([analyticsMock isEnabled]).andReturn(NO);
+  [MSAppCenter configureWithAppSecret:kMSTestAppSecret];
+  [[MSAnalytics sharedInstance] startWithLogManager:logManagerMock appSecret:kMSTestAppSecret];
+
+  // FIXME: logManager holds session tracker somehow and it causes other test failures. Stop it for hack.
+  [[MSAnalytics sharedInstance].sessionTracker stop];
+
+  // When
+  OCMReject([logManagerMock processLog:OCMOCK_ANY forGroupId:OCMOCK_ANY]);
+  [[MSAnalytics sharedInstance] trackEvent:@"Some event" withProperties:nil];
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: You can also verify `validateEventName:forLogType:` and `validateProperties:forLogName:andType:`  not being called.

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m [hunk 3]
```diff
@@ -329,6 +365,45 @@ - (void)testTrackEventWithoutProperties {
...
+  id logManagerMock = OCMProtocolMock(@protocol(MSLogManager));
+  [MSAppCenter configureWithAppSecret:kMSTestAppSecret];
+  [[MSAnalytics sharedInstance] startWithLogManager:logManagerMock appSecret:kMSTestAppSecret];
+
+  // FIXME: logManager holds session tracker somehow and it causes other test failures. Stop it for hack.
+  [[MSAnalytics sharedInstance].sessionTracker stop];
+
+  // When
+  OCMReject([logManagerMock processLog:OCMOCK_ANY forGroupId:OCMOCK_ANY]);
+  [[MSAnalytics sharedInstance] trackEvent:invalidEventName withProperties:nil];
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: Add more verifies here.
`validateEventName:forLogType:`, `validateProperties:forLogName:andType:` should be called in this test.

File: AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m [hunk 4]
```diff
@@ -238,7 +238,7 @@ - (void)trackEvent:(NSString *)eventName withProperties:(NSDictionary<NSString *
 }
 
 - (void)trackPage:(NSString *)pageName withProperties:(NSDictionary<NSString *, NSString *> *)properties {
-  if (![super isEnabled])
+  if (![self isEnabled])
```
Comment (TroubleMakerBen) [inline AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m]: Good catch!

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m (hunk 2 above)
Comment (jaelim-ms) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: Same for trackPage test

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m (hunk 3 above)
Comment (jaelim-ms) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: Same for trackPage test

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m (hunk 1 above)
Comment (bmourat) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: I'm testing the case where autoPageTracking is enabled and missedPageViewName returns not nil, then trackPage should be called with missedPageViewName.

Comment (TroubleMakerBen) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: This makes sense to me.

Comment (jaelim-ms) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: The reason for the comment is that the method name is not quite descriptable for its testing. Can you rename?

**Pull request merged**